service/mapper/transitions.go

```diff
@@ -141,12 +143,17 @@ func (t *Transitions) UpdateTree(s *State) error {
 	// it to in the forest. This usually means that it was meant for a pruned
 	// branch of the execution forest.
 	update, err := t.feed.Update()
+	if err == dps.ErrUnavailable {
+		time.Sleep(t.cfg.WaitInterval)
+		return nil
+	}
```

Comment:
Why remove the waiting state, in the end? Is it because we now have two different components that can return an ErrUnavailable?

Reply:
Yes, I didn't like the redundancy. I prefer this over having to other have to transitions that are exactly the same code, or having to add some additional state to know which transition to transition back into.